src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -1251,6 +1267,137 @@ static void add_stream_kafka_command(int connFd, std::string &kafka_server_IP, c
 }
 
+void add_stream_hdfs_command(std::string masterIP,int connFd, std::string &hdfs_server_IP, std::thread &input_stream_handler_thread,
+                             std::vector<DataPublisher *> &workerClients, int numberOfPartitions,
+                             SQLiteDBInterface *sqlite, bool *loop_exit_p) {
+    std::string hdfs_port;
+    std::string msg_1 = "Do you want to use the default HDFS server(y/n)?";
+    int result_wr = write(connFd, msg_1.c_str(), msg_1.length());
+    if (result_wr < 0) {
+        frontend_logger.error("Error writing to socket");
+        *loop_exit_p = true;
+        return;
+    }
+    result_wr = write(connFd, "\r\n", 2);
+    if (result_wr < 0) {
+        frontend_logger.error("Error writing to socket");
+        *loop_exit_p = true;
+        return;
+    }
+
+    char user_res[FRONTEND_DATA_LENGTH + 1];
+    bzero(user_res, FRONTEND_DATA_LENGTH + 1);
+    read(connFd, user_res, FRONTEND_DATA_LENGTH);
+    std::string user_res_s(user_res);
+    user_res_s = Utils::trim_copy(user_res_s);
+    for (char &c: user_res_s) {
+        c = tolower(c);
+    }
+
+    if (user_res_s == "y") {
+        hdfs_server_IP = Utils::getJasmineGraphProperty("org.jasminegraph.server.streaming.hdfs.host");
+        hdfs_port = Utils::getJasmineGraphProperty("org.jasminegraph.server.streaming.hdfs.port");
+    } else {
+        std::string message = "Send the file path to the HDFS configuration file.";
+        result_wr = write(connFd, message.c_str(), message.length());
+        if (result_wr < 0) {
+            frontend_logger.error("Error writing to socket");
+            *loop_exit_p = true;
+            return;
+        }
+        result_wr = write(connFd, "\r\n", 2);
+        if (result_wr < 0) {
+            frontend_logger.error("Error writing to socket");
+            *loop_exit_p = true;
+            return;
+        }
+
+        char file_path[FRONTEND_DATA_LENGTH + 1];
+        bzero(file_path, FRONTEND_DATA_LENGTH + 1);
+        read(connFd, file_path, FRONTEND_DATA_LENGTH);
+        std::string file_path_s(file_path);
+        file_path_s = Utils::trim_copy(file_path_s);
+
+        frontend_logger.info("Reading HDFS configuration file: " + file_path_s);
+
+        std::vector<std::string> vec = Utils::getFileContent(file_path_s);
+        for (const auto &item : vec) {
+            frontend_logger.info("Item: " + item);
+            if (item.length() > 0 && !(item.rfind("#", 0) == 0)) {
+                std::vector<std::string> vec2 = Utils::split(item, '=');
+                if (vec2.size() == 2) {
+                    if (vec2.at(0).compare("hdfs.host") == 0) {
+                        hdfs_server_IP = vec2.at(1);
+                    } else if (vec2.at(0).compare("hdfs.port") == 0) {
+                        hdfs_port = vec2.at(1);
+                    }
+                } else {
+                    frontend_logger.error("Invalid line in configuration file: " + item);
```

Comment:
Which configuration file are you referring to here?

Reply:
In this hdfs implementation, there are two types of HDFS servers.
1. Jasminegraph's Default HDFS server.
2. Custom HDFS servers from users.
users can use either one of those two types. if a user wants to refer to a custom hdfs server, then he needs to provide config file to that HDFS server. Here it refers to the custom hdfs server config file.